content/blog/the-rise-of-special-project-infra.md

```diff
@@ -0,0 +1,168 @@
+
+While the implications of endorsing such platforms remain all but imperceptible to the general public, for Summer of Nix (SoN) participants—each of whom contractually agreed with Eelco himself to uphold both [NGI Zero](https://nlnet.nl/NGI0/)’s aim to contribute to an open internet and the [ACM Code of Ethics and Professional Conduct](https://www.acm.org/code-of-ethics), which emphasizes among other points the importance of respecting privacy—streaming the inaugural event in the premier SoN Public Lecture Series exclusively to closed-source, centralized, and infamously privacy-disrespecting services was an irony too plain to ignore.
+
+Fortunately, infrastructural improvement is a fundamental objective of the SoN program—and Nix is the definitive tool for the job—making the Public Lecture Series livestreaming infrastructure a natural first target. Time was not on our side, but good people were:
+
+- The [NixOS Foundation](https://opencollective.com/nixos) was at the ready for server provisioning and DNS management.
+- Personnel at [Tweag](https://www.tweag.io/) generously offered expertise in configuration and deployment in a pinch.
+- A new SoN Special Project—tidily dubbed _infra_—was launched to formalize the effort and attract participation.
+
+With this kind of support, the odds were decidedly stacked in our favor, _but could we deliver?_ We’d find out soon enough...
+
+### Phase I: Research
+
+Everybody needs a saga. Ours was: self-host a secure livestreaming server to accommodate a sizable audience participating from all around the world and at variable network speeds, _fast_. Oh, and add a chat room for synchronous Q&A.
+
+On the server side, we settled on the following stack: virtual private server (VPS) → NixOS → [Caddy](https://caddyserver.com/) → [Owncast](https://owncast.online/).
+
+The media would originate on our intrepid moderator’s local [OBS Studio](https://obsproject.com/) and be streamed at archival quality over the Real-Time Messaging Protocol (RTMP) to our remote Owncast service. Owncast would transcode the video to multiple stream qualities (i.e., optimized variants of video bitrate and CPU usage) and present the selected variant to the user in an attractive, branded web UI with a convenient chat box.
+
+### Phase II: Provision
+
+Controlling the narrative has always called for a bit of hardware. For this exercise, we went with a single VPS equipped with dedicated vCPUs. The general provisioning procedure follows:
```

Comment:
> Controlling the narrative has always called for a bit of hardware.

I'm not sure what that sentence is supposed to mean. Which narrative?

Reply:
The expression _controlling the narrative_ (or _framing the discourse_, as it’s sometimes phrased) is used here as a set phrase. In the context of self-hosting, this implies maintaining ownership of both content and its presentation, thereby offering users a way to engage with said content free from third-party influence.

As both the expression and the term _hardware_ are open to interpretation, however, identification of subtext is an exercise left for the reader.

Reply:
I almost supposed that, but it's not as self-explanatory as "framing the narrative", which would actually play nicer with the hardware part: framing the narrative with a bit of hardware.

Reply:
👍